pylint/checkers/utils.py

```diff
@@ -1251,6 +1251,19 @@ def safe_infer(node: nodes.NodeNG, context=None) -> Optional[nodes.NodeNG]:
             inferred_type = _get_python_type_of_node(inferred)
             if inferred_type not in inferred_types:
                 return None  # If there is ambiguity on the inferred node.
+            if isinstance(inferred, astroid.FunctionDef):
+                # Special case due to unexpected inference ambiguity
+                if (
+                    inferred.name == "TemporaryFile"
```

Comment:
I think this is very specific to the issue with tempfile, probably too much.

Reply:
I'll try to diagnose the issue, maybe it's in astroid.